ui/app/utils/parse-pki-cert.js

```diff
@@ -131,15 +131,29 @@ export async function verifyCertificates(certA, certB, leaf) {
   const parsedCertB = jsonToCertObject(certB);
   if (leaf) {
     const parsedLeaf = jsonToCertObject(leaf);
-    const chainA = await parsedLeaf.verify(parsedCertA);
-    const chainB = await parsedLeaf.verify(parsedCertB);
+    const chainA = await verifySignature(parsedCertA, parsedLeaf);
+    const chainB = await verifySignature(parsedCertB, parsedLeaf);
     // the leaf's issuer should be equal to the subject data of the intermediate certs
     const isEqualA = parsedLeaf.issuer.isEqual(parsedCertA.subject);
     const isEqualB = parsedLeaf.issuer.isEqual(parsedCertB.subject);
     return chainA && chainB && isEqualA && isEqualB;
   }
   // can be used to validate if a certificate is self-signed (i.e. a root cert), by passing it as both certA and B
-  return (await parsedCertA.verify(parsedCertB)) && parsedCertA.issuer.isEqual(parsedCertB.subject);
+  return (await verifySignature(parsedCertA, parsedCertB)) && parsedCertA.issuer.isEqual(parsedCertB.subject);
+}
+
+async function verifySignature(parent, child) {
+  try {
+    // ed25519 is an unsupported signature algorithm
+    // catch the error and instead check the AKID (authority key ID) includes the SKID (subject key ID)
+    return await child.verify(parent);
```

Comment:
We don't need `await` here, we can just return `child.verify` directly and even remove `async` on the method since we're just returning the promise

Reply:
ah yes! thanks